rust/src/main.rs

```diff
@@ -42,22 +42,22 @@ impl EvTag {
         let object = odb.read(object.id())?;
         let contentbuf = object.data();
         let header = format!("{} {}", object.kind().str(), contentbuf.len());
-        hash.write(header.as_bytes()).unwrap();
+        hash.write_all(header.as_bytes()).unwrap();
         let nulbyte: [u8; 1] = [0; 1];
-        hash.write(&nulbyte).unwrap();
-        hash.write(contentbuf).unwrap();
+        hash.write_all(&nulbyte).unwrap();
+        hash.write_all(contentbuf).unwrap();
         Ok(())
     }
 
     fn checksum_submodule(&self, hash: &mut Hasher, submodule: &Submodule) -> Result<(), Error> {
         let subrepo = submodule.open()?;
         let sub_head = submodule.workdir_id().expect("Failed to find workdir id");
         let commit = subrepo.find_commit(sub_head)?;
-        self.checksum_commit_contents(&subrepo, commit, hash)?;
+        self.checksum_commit_contents(&subrepo, &commit, hash)?;
```

Comment:
Hum interesting.  Is this like "unnecessary move" or something?

Reply:
I think it was https://rust-lang-nursery.github.io/rust-clippy/master/index.html#needless_pass_by_value

Reply:
Ah yep.  That's a cool lint.